internal/services/cosmos/cosmosdb_cassandra_datacenter_resource.go

```diff
@@ -195,63 +191,65 @@ func resourceCassandraDatacenterRead(d *pluginsdk.ResourceData, meta interface{}
 		return fmt.Errorf("reading %q: %+v", id, err)
 	}
 
-	clusterId := parse.NewCassandraClusterID(id.SubscriptionId, id.ResourceGroup, id.CassandraClusterName)
+	clusterId := managedcassandras.NewCassandraClusterID(id.SubscriptionId, id.ResourceGroupName, id.CassandraClusterName)
 	d.Set("name", id.DataCenterName)
 	d.Set("cassandra_cluster_id", clusterId.ID())
-	if props := resp.Properties; props != nil {
-		if res := props; res != nil {
-			d.Set("delegated_management_subnet_id", props.DelegatedSubnetID)
-			d.Set("location", location.NormalizeNilable(props.DataCenterLocation))
-			d.Set("backup_storage_customer_key_uri", props.BackupStorageCustomerKeyURI)
-			d.Set("base64_encoded_yaml_fragment", props.Base64EncodedCassandraYamlFragment)
-			d.Set("managed_disk_customer_key_uri", props.ManagedDiskCustomerKeyURI)
-			d.Set("node_count", props.NodeCount)
-			d.Set("disk_count", int(*props.DiskCapacity))
-			d.Set("disk_sku", props.DiskSku)
-			d.Set("sku_name", props.Sku)
-			d.Set("availability_zones_enabled", props.AvailabilityZone)
+	if model := resp.Model; model != nil {
+		if props := model.Properties; props != nil {
+			if res := props; res != nil {
+				d.Set("delegated_management_subnet_id", props.DelegatedSubnetId)
+				d.Set("location", location.NormalizeNilable(props.DataCenterLocation))
+				d.Set("backup_storage_customer_key_uri", props.BackupStorageCustomerKeyUri)
+				d.Set("base64_encoded_yaml_fragment", props.Base64EncodedCassandraYamlFragment)
+				d.Set("managed_disk_customer_key_uri", props.ManagedDiskCustomerKeyUri)
+				d.Set("node_count", props.NodeCount)
+				d.Set("disk_count", int(*props.DiskCapacity))
+				d.Set("disk_sku", props.DiskSku)
+				d.Set("sku_name", props.Sku)
+				d.Set("availability_zones_enabled", props.AvailabilityZone)
+			}
```

Comment:
this `res` assignment isn't needed?

```suggestion
			d.Set("delegated_management_subnet_id", props.DelegatedSubnetId)
			d.Set("location", location.NormalizeNilable(props.DataCenterLocation))
			d.Set("backup_storage_customer_key_uri", props.BackupStorageCustomerKeyUri)
			d.Set("base64_encoded_yaml_fragment", props.Base64EncodedCassandraYamlFragment)
			d.Set("managed_disk_customer_key_uri", props.ManagedDiskCustomerKeyUri)
			d.Set("node_count", props.NodeCount)
			d.Set("disk_count", int(*props.DiskCapacity))
			d.Set("disk_sku", props.DiskSku)
			d.Set("sku_name", props.Sku)
			d.Set("availability_zones_enabled", props.AvailabilityZone)
```

Reply:
Updated